monailabel/endpoints/datastore.py

```diff
@@ -181,6 +182,28 @@ def update_label_info(label: str, tag: str, info: str = Form("{}"), user: Option
     return instance.datastore().update_label_info(label, tag, i)
 
 
+def download_dataset(limit_cases: Optional[int] = None):
+    instance: MONAILabelApp = app_instance()
+    dl = instance.datastore().datalist()
+
+    if not len(dl) > 0:
+        return {}
+
+    if limit_cases and limit_cases in list(range(1, len(dl))):
+        logger.info(f"Number of cases in datalist reduced to: {limit_cases} of {len(dl)} case(s)")
+        dl = dl[:limit_cases]
+
+    with zipfile.ZipFile("dataset.zip", mode="w") as archive:
+        logger.info(f"ZIP archive will be written to: {archive.filename}")
+        for d in dl:
+            # write image and corresponding label file to archive
+            for key in d.keys():
+                path = d[key]
+                archive.write(path, arcname=os.path.join(key, os.path.basename(path)))
+    assert archive.filename is not None
+    return FileResponse(archive.filename, filename=os.path.basename(archive.filename))
+
+
 @router.get("/", summary="Get All Images/Labels from datastore")
```

Comment:
Option 1::
GET /datatstore?include_images=False
(We can use the existing API with additional URI parameter to define this instead of introducing new API)

Option 2::
GET /datatstore/dataset as new API to download the images+labels+dataset.json for research purposes 

Current name is little confusing 
GET /datatstore/downloadlabeleddataset

I feel option 2 is better.. later variations can be added to same endpoint to support different formats etc..